### ISSUE TITLE
Rename window does not contain right text when cursor is between unrelated quotes.

### ISSUE THREAD

**Issue by qutreson** [Microsoft/vscode#48137]
**Version:**
- VSCode Version: 1.23.0-insider - 88f500e805ecc4efd064f7886c1a5ce8a2a2c087
- OS Version: Win10

**Problem:**
When renaming an identifier that is in between two other ones with quotes like `Identifier2` in `"Identifier 1".Identifier2 where "Identifier 3".`, the text field that pops up does not contain only this identifier, but all the text in between quotes (here, `.Identifier2 where "`).

Below is an example from the AL extension:

![vscoderenamingissue](https://user-images.githubusercontent.com/31541422/38938728-8a0044b0-4326-11e8-874d-0dcb11926e89.gif)

**Note:** As you can see in the gif, this bug only concerns the text field which opens. After removing the text, the renaming works perfectly (also the span for replacement only covers the right name and not the text display at first which is good).

**Steps to Reproduce:**

1. Install AL extension from marketplace
2. Create a AL project using command AL:Go. You will be request to chose a server, but you don't need it to reproduce the problem here.
3. In the .al file created, Copy-Paste the following code:
```
table 50100 "My Table"
{
    fields
    {
        field(1; "My Field"; Integer) { }
        field(2; ToRename; Text[50]) { }
    }
}

table 50101 MyTable
{
    fields
    {
        field(2; MyNormalField; Code[10]) { }
        field(3; MyFlowField; Text[5])
        {
            FieldClass = FlowField;           
            CalcFormula = lookup ("My Table".To^Rename where ("My Field" = const (5)));
        }
    }
}
```
4. Remove the ^ and trigger Renaming at its position.

**Expected:** The text field pops up with `ToRename`.
**Actual:** The text field contains all the text between quotes: `".ToRename where ("`

**Comment by ramya-rao-a** [comment]
@jrieken I believe this is due to the range as determined by the rename action?

**Comment by qutreson** [comment]
Can you please add the issue it is a duplicate of so that I can follow the thread?

**Comment by qutreson** [comment]
@jrieken , @ramya-rao-a, any indication on which thread to follow for this bug?

**Comment by ramya-rao-a** [comment]
@qutreson #7340 tracks the feature request for an extension to be able to provide the range of the text to be renamed which can then be used by VS Code to provide the rename window. In the absence of this information VS Code will try to use the word definition to determine the range.

Please log a feature request in the repository of the extension you use for `al` files to use this new API which will be available in the upcoming release of 1.23
